events/events.go

```diff
@@ -31,9 +35,83 @@ type EventListener interface {
 	HandleEvent(event Event)
 }
 
+// EventEmitter describes an interface that can be used to emit events
+type EventEmitter interface {
+	EmitEvent(Event)
+}
+
 // EventRegistrar is an object that you can register EventListeners on.
 type EventRegistrar interface {
 	RegisterListener(EventListener)
+	DeregisterListener(EventListener)
+}
+
+type baseEventRegistrar struct {
+	lock      sync.RWMutex
```

Comment:
By embedding it in the struct the functions from `RWMutex` will become part of its public interface which is undesirable. Not only does it add to the public interface it might also make it hard for the event emitting struct to be embedded in an other struct due to s collision in functions.

This lock is now specifically private to the event registrar that embeds the base.